### DISCUSSION TITLE
Do you clone this repo into themes/tailbliss or just clone it as mysitename and edit everything?

### DISCUSSION THREAD

**Discussion by peterarsenault-at** [nusserstudios/tailbliss#171] [General]
Hi, I'm a bit confused using Tailbliss, possibly because I've never used a theme with node.js before. As I understand Hugo, the standard procedure for using a theme is to [submodule it](https://gohugo.io/getting-started/quick-start/) in themes/_themename_ and add a "theme = _'themename'"_ in hugo.toml.

Do I understand correctly that with tailbliss, you don't bother with that, but rather clone it as _mysitename_ and then, rather than running `hugo serve`, you instead run all commands through NPM? This would mean no submodules, and just directly change all the content in-situ? 

I suppose this would mean that the deployment script (e.g., amplify.yml) would need commands for both installing hugo and installing npm. I plan to play around with the deployment scripts, but I am mostly curious about the first point.

**Comment by KhalilSecurity** [comment]
You are right, there is no submodule. Start by installing hugo extended using snap or similar, then:

1. Clone the repo
2. cd tailbliss
3. sudo apt install npm
4. sudo snap install go --classic
5. sudo npm install
6. sudo npm run build

To test if it's working (change to your PC IP)
sudo hugo server --bind 192.168.217.24 --baseURL http://192.168.217.24 

Make your edits as you wish, and don't forget the config here is hugo.yaml not .toml (format of yaml is way better anyway)
To generate /public to upload to your website
sudo hugo --gc --minify

**Reply by SylwiaNowakowska** [reply]
Hi,

Thx a lot for this instruction. 

I am working WSL and made the following steps:
- sudo apt install hugo
- git clone https://github.com/nusserstudios/tailbliss
- cd tailbliss
- sudo apt install npm
- installed go by: sudo apt install golang-go
- sudo apt npm install

I have a problem with the last step:
- sudo npm run build

> TailBliss@0.1.1 build
> hugo --minify

Error: Unable to locate config file or config directory. Perhaps you need to create a new site.
       Run `hugo help new` for details.

I would appreciate your help!

**Reply by SylwiaNowakowska** [reply]
Running: hugo server --config hugo.yaml solved the issue :)

**Comment by SnapshotCiTy** [comment]
do those instructions work on macOS as well?

**Reply by SnapshotCiTy** [reply]
Sorry i'm very inexperienced with JavaScript and npm. Trying to use the tailwind theme with Hugo, trying to figure things out.

on macOS ventura, after a standard hugo installation, did:
`brew install npm`
`npm run build`
- resulted in an error, postcss-cli was missing
`npm install post-css-cli`

`npm run build`

`npm run start`seem to build and start the hugo server.

Question 1: does it mean that we can only build with `npm run build`and serve it locally with `npm run start`and NOT `hugo` or `hugo server`?

Question 2: did `npm install -D @tailwindcss/typography` and confirmed that the `plugins: [require("@tailwindcss/typography")]` is in the `tailwind.config.js`file. And it is... But I'm not sure that the 'typography' features of tailwind are active, the 'typography replacements' shown on the example About page, do now seem to work. What am I missing?

Thanks in advance to anyone willing to help.

**Reply by nusserstudios** [reply]
Yes, it works fine on MacOS. I'd suggest using NVM (node Version manager with brew). This works with node v20 fine. npm install works fine. I'd even recommend pnpm. Its a lot faster. Install Hugo with Brew as well. Once you have those two, it should compile just fine. https://formulae.brew.sh/formula/nvm https://formulae.brew.sh/formula/hugo#default. Version 18 should work too. Commands: npm i, npm run build, npm run start will get you going. You can put http://localhost:1313/ as the Hugo BaseURL in hugo.yaml file.
